### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by rbd3**
# Change made :
👉 Make login required
👉 Use devise gem for authentication

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (danmainah) [inline app/views/devise/sessions/new.html.erb:17]: - [ ] @Team Great work adding the login page, however the design slightly differs from the design given, check the comparison below:

Given Design          |  Your design     
:-------------------------:|:-------------------------:
image | image

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb]: - [ ] @Team Great work adding the registration page, however the design slightly differs from the design given , check the comparison below:

Given Design          |  Your design     
:-------------------------:|:-------------------------:
image | image

Comment (danmainah) [reply app/views/devise/registrations/new.html.erb]: - [ ] @Team I was unable to sign up as a new user even after filling are the fields in the form, am getting the error below, Kindly consider checking your devise setup to ensure the login and registration work correctly.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @rbd3 and @noelfoka , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Add devise gem and setup it up.
✔️ Modify the Signup and Login views to match the design given.

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by binhussen — APPROVED** [review]
Hi @rbd3 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
IApproveSimonCowellGIF (6)

## Highlights
- Sticked Given wireframe ✔️
- Well Stractured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**